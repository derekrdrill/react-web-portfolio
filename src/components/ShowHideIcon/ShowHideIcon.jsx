import React, { useState } from 'react';
import { EyeIcon } from './components/EyeIcon';

export const ShowHideIcon = ({ onClick }) => {
  const [locked, setLocked] = useState(true);
  const handleSetLocked = () => setLocked(!locked);

  return (
    <EyeIcon
      locked={locked}
      onClick={() => {
        handleSetLocked();
        onClick && onClick();
      }}
    />
  );
};


