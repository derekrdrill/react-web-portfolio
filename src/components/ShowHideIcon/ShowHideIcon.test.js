import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import { ShowHideIcon, EyeIcon } from './ShowHideIcon';

const renderer = new ShallowRenderer();

describe('Show Hide Icon tests', () => {
  it('renders correctly', () => {
    renderer.render(<ShowHideIcon />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });

  it('renders eye icon', () => {
    const eyeIconTests = [
      { locked: true, result: 'eye' },
      { locked: false, result: 'eye-slash' },
    ];

    eyeIconTests.forEach(({ locked, result }) => {
      renderer.render(<EyeIcon locked={locked} />);
      const render = renderer.getRenderOutput();
      const iconName = render.props.icon.iconName;

      expect(iconName).toEqual(result);
    });
  });
});
