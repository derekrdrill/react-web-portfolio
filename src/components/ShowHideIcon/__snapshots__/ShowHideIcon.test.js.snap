// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show Hide Icon tests renders correctly 1`] = `
<EyeIcon
  locked={true}
  onClick={[Function]}
/>
`;
